components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/models/BFSIRequestObject.java

```diff
@@ -0,0 +1,142 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.identity.extensions.auth.extensions.request.validator.models;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.PlainJWT;
+import com.nimbusds.jwt.SignedJWT;
+import org.wso2.bfsi.identity.extensions.validator.annotation.RequiredParameter;
+import org.wso2.bfsi.identity.extensions.validator.annotation.RequiredParameters;
+import org.wso2.bfsi.identity.extensions.validator.annotation.ValidAudience;
+import org.wso2.bfsi.identity.extensions.validator.annotation.ValidScopeFormat;
+import org.wso2.bfsi.identity.extensions.validator.annotation.ValidSigningAlgorithm;
+import org.wso2.carbon.identity.oauth2.RequestObjectException;
+import org.wso2.carbon.identity.openidconnect.model.RequestObject;
+import org.wso2.carbon.identity.openidconnect.model.RequestedClaim;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A decorator class of RequestObject to enforce validations. Can add delegate methods as required.
+ *
+ * @param <T> Any child of this class.
+ */
+@RequiredParameters({
+        @RequiredParameter(param = "signedJWT",
+                message = "Only Signed JWS is accepted for request object"),
+        @RequiredParameter(param = "claimsSet.claims.aud",
+                message = "aud parameter is missing in the request object")
+})
+@ValidScopeFormat(scope = "claimsSet.claims.scope")
+@ValidAudience(audience = "claimsSet.claims.aud", clientId = "claimsSet.claims.client_id")
+@ValidSigningAlgorithm(algorithm = "signedJWT.header.algorithm.name", clientId = "claimsSet.claims.client_id")
+public class BFSIRequestObject<T extends BFSIRequestObject> extends RequestObject {
+
+    // decorator object
+    private RequestObject requestObject;
+    private static final long serialVersionUID = -568639546792395972L;
+
+    public BFSIRequestObject(RequestObject requestObject) throws RequestObjectException {
+        if (requestObject == null) {
+            throw new RequestObjectException(RequestObjectException.ERROR_CODE_INVALID_REQUEST,
+                    "Null request object passed");
+        }
+        this.requestObject = requestObject;
+    }
+
+
+    // for tool kits to use
+
+    /**
+     * Any child object of this class can create object of this class using this constructor.
+     *
+     * @param childObject Any class extending this class.
+     */
+    public BFSIRequestObject(T childObject) {
```

Comment:
Why cant we use RequestObject type here instead of T?

Reply:
We want allow any extended implementation of RequestObject to be passed to the method with newly added attributes